src/renderer/uia/UiaRenderer.cpp

```diff
@@ -189,7 +216,12 @@ UiaEngine::~UiaEngine()
     {
         _isPainting = false;
 
-        // TODO CARLOS: fire UIA events
+        // Fire UIA events here
+        if (_selectionChanged)
+        {
+            _dispatcher->SignalUia(ConsoleUiaEvent::SelectionChanged);
+            _selectionChanged = false;
+        }

```

Comment:
Should this be done during the `StartPaint` or the `EndPaint`?

Reply:
Actually, I was just thinking that maybe this should be moved to `PaintSelection`. That way each event is in it's own context. Thoughts?

Reply:
Ended up moving it to `EndPaint`. I think it makes more sense even though nothing is done during the regular `Paint` steps.

When I was trying to do this in the `Paint` steps, I ran into a problem of the renderer _needing_ the rects for what changed. We don't need to go through the effort of finding/calculating the rects for each row, so I decided to just make all of them return `S_FALSE` instead and leave it up to my private `bool` to decide if we should actually signal or not.